x-pack/plugins/alerting/server/saved_objects/migrations.ts

```diff
@@ -467,6 +476,89 @@ function removeNullAuthorFromSecurityRules(
   };
 }
 
+/**
+ * This migrates exception list containers to saved object references on an upgrade.
+ * We only migrate if we find these conditions:
+ *   - exceptionLists are an array and not null, undefined, or malformed data.
+ *   - The exceptionList item is an object and id is a string and not null, undefined, or malformed data
+ *   - The existing references do not already have an exceptionItem reference already found within it.
+ * Some of these issues could crop up during either user manual errors of modifying things, earlier migration
+ * issues, etc...
+ * @param doc The document that might have exceptionListItems to migrate
+ * @returns The document migrated with saved object references
+ */
+function addExceptionListsToReferences(
+  doc: SavedObjectUnsanitizedDoc<RawAlert>
+): SavedObjectUnsanitizedDoc<RawAlert> {
+  const {
+    attributes: {
+      params: { exceptionsList },
+    },
+    references,
+  } = doc;
+  if (!Array.isArray(exceptionsList)) {
+    // early return if we are not an array such as being undefined or null or malformed.
+    return doc;
+  } else {
+    const exceptionsToTransform = removeMalformedExceptionsList(exceptionsList);
+    const newReferences = exceptionsToTransform.flatMap<SavedObjectReference>(
+      (exceptionItem, index) => {
+        const existingReferenceFound = references?.find((reference) => {
+          return reference.id === exceptionItem.id;
```

Comment:
Should this also check for reference.type = 'exception-list' in the (admittedly unlikely event) of another SO reference with the same ID but different type?

Reply:
Yes, the tighter I can lock this down the better.